src/gamma_inc.jl

```diff
@@ -782,112 +787,121 @@ See also [`gamma(z)`](@ref SpecialFunctions.gamma), [`gamma_inc_inv(a,p,q)`](@re
+                    throw(DomainError((a, x, ind), "P(a,x) or Q(a,x) is computationally indeterminant in this case."))
                 end
                 c = exp(-y)
                 w = 0.5*erfcx(sqrt(y))
                 u = 1.0/a
                 z = sqrt(z + z)
                 if l < 1.0
-                    z=-z
+                    z = -z
                 end
                 if iop == 1
-                    return gamma_inc_minimax(a,x,z)
+                    return gamma_inc_minimax(a, x, z)
                 elseif iop == 2
-                    return gamma_inc_temme(a,x,z)
+                    return gamma_inc_temme(a, x, z)
                 else
-                    t = @horner(z , d00 , d0[1] , d0[2] , d0[3])
+                    t = @horner(z, d00, d0[1], d0[2], d0[3])
                     return gamma_inc_temme_1(a, x, z, ind)
                 end
-            end
-        elseif a > x || x >= x0[iop] || !isinteger(2*a)
-            r = rgammax(a,x)
-            if r == 0.0
-                if x <= a
-                    return (0.0,1.0)
-                else
-                    return (1.0,0.0)
-                end
-            end
-            if x <= max(a,alog10)
-                return gamma_inc_taylor(a, x, ind)
-            elseif x < x0[iop]
-                return gamma_inc_cf(a, x, ind)
             else
-                return gamma_inc_asym(a, x, ind)
+                return _gamma_inc_choose_algorithm(a, x, ind)
             end
+        elseif a > x || x >= x0[iop] || !isinteger(2*a)
+            return _gamma_inc_choose_algorithm(a, x, ind)
         else
             return gamma_inc_fsum(a,x)
-
         end
     elseif a == 0.5
         if x >= 0.25
             q = erfc(sqrt(x))
-            return ( 1.0 - q , q )
+            return (1.0 - q, q)
         end
         p = erf(sqrt(x))
-        return ( p , 1.0 - p )
+        return (p, 1.0 - p)
     elseif x < 1.1
         return gamma_inc_taylor_x(a, x, ind)
     end
-    r = rgammax(a,x)
+    r = rgammax(a, x)
     if r == 0.0
         return (1.0, 0.0)
     else
         return gamma_inc_cf(a, x, ind)
     end
-
-
-
 end
 
 function _gamma_inc(a::BigFloat,x::BigFloat,ind::Integer) #BigFloat version from GNU MPFR wrapped via ccall
     z = BigFloat()
-    ccall((:mpfr_gamma_inc, :libmpfr), Int32 , (Ref{BigFloat} , Ref{BigFloat} , Ref{BigFloat} , Int32) , z , a , x , ROUNDING_MODE[])
+    ccall((:mpfr_gamma_inc, :libmpfr), Int32, (Ref{BigFloat}, Ref{BigFloat}, Ref{BigFloat}, Int32), z, a, x, ROUNDING_MODE[])
     q = z/gamma(a)
     return (1.0 - q, q)
 end
 _gamma_inc(a::Float32,x::Float32,ind::Integer) = Float32.(_gamma_inc(Float64(a),Float64(x),ind))
 _gamma_inc(a::Float16,x::Float16,ind::Integer) = Float16.(_gamma_inc(Float64(a),Float64(x),ind))
 
+function _gamma_inc_choose_algorithm(a::Float64, x::Float64, ind::Int)
+    r = rgammax(a, x)
+    if r == 0.0
+        # It shouldn't be possible to reach this point if a >= x
+        @assert a < x
```

Comment:
Can you explain why this is not possible? It seems in the Fortran code there are two branches for `x <= a` (331) and `x > a` (the line below).

Reply:
I tried to construct a test case to hit that branch from `_gamma_inc` and I had convinced myself that it wasn't possible but I tried tried again and now I succeed
```julia
julia> gamma_inc(24.0, 1e-323)
ERROR: AssertionError: a < x
```
so I'll update this part and add the test.